### PR TITLE
Remove leading zeroes in prerelease components.

### PR DESCRIPTION
**Pull request by chriseth**

### CODE REVIEW

File: cmake/scripts/buildinfo.cmake [hunk 1]
```diff
@@ -26,6 +26,7 @@ if (EXISTS ${ETH_SOURCE_DIR}/prerelease.txt)
 	string(STRIP "${SOL_VERSION_PRERELEASE}" SOL_VERSION_PRERELEASE)
 else()
 	string(TIMESTAMP SOL_VERSION_PRERELEASE "develop.%Y.%m.%d" UTC)
+        string(REPLACE .0 . SOL_VERSION_PRERELEASE "${SOL_VERSION_PRERELEASE}")
```
Comment (pirapira) [inline cmake/scripts/buildinfo.cmake]: The indent seems to have changed, though I don't know the consequences in `cmake` files.

**Comment by chriseth** [comment]
Sorry, I need to merge this now, otherwise browser-solidity will keep on breaking.

File: scripts/travis-emscripten/publish_binary.sh [hunk 2]
```diff
@@ -34,7 +34,10 @@ VER=$(cat CMakeLists.txt | grep 'set(PROJECT_VERSION' | sed -e 's/.*set(PROJECT_
 test -n "$VER"
 VER="v$VER"
 COMMIT=$(git rev-parse --short HEAD)
-DATE=$(date --date="$(git log -1 --date=iso --format=%ad HEAD)" --utc +%Y.%m.%d)
+DATE=$(date --date="$(git log -1 --date=iso --format=%ad HEAD)" --utc +%Y.%-m.%-d)
+
+# remove leading zeros in components - they are not semver-compatible
+COMMIT=$(echo "$COMMIT" | sed -e 's/^0*//')
```
Comment (ekpyron) [inline scripts/travis-emscripten/publish_binary.sh:40]: Was it wrong to add this here or was this part intentional as well? This is basically what introduces https://github.com/ethereum/solidity/issues/8127

Comment (chriseth) [reply scripts/travis-emscripten/publish_binary.sh:40]: It was intentional. There cannot be leading zeros in this semver component.

**Pull request merged**